backend/coreapp/views/scratch.py

```diff
@@ -427,6 +435,58 @@ def export(self, request: Request, pk):
             },
         )
 
+    @action(detail=True, methods=["POST"])
+    def pr(self, request: Request, pk):
+        scratch: Scratch = self.get_object()
+        if scratch.owner != request.profile:
+            raise ScratchNotOwnerException()
+        # TODO: catch exceptions
+        fn: ProjectFunction = scratch.project_function
+        token = request.profile.user.github.access_token
+        github_repo: Repository = fn.project.repo.details(token)
+
+        # Get or create fork
+        try:
+            # TODO: likely need to pull this to make it up to date
+            fork = Github(token).get_repo(f"{scratch.owner}/{github_repo.name}")
+        except UnknownObjectException:
+            # TODO: maybe just call this? POST == GET if exists?
+            fork = Github(token).get_user().create_fork(github_repo)
+
+        # Change file contents
+        contents = fork.get_contents(fn.src_file)
+        new_content = re.compile(
+            # TODO: escape function name?
+            rf"INCLUDE_ASM\([^,]+, [^,]+, {fn.display_name}[^\)]*\);",
```

Comment:
This should be the diff_label of the function probably. There's also no point don't `re.compile` if we're gonna do `.sub` after; just do `re.sub()`

Reply:
What are the differences between `ProjectFunction.diff_label` and `Scratch.display_name`? This is used in three places:
1. Here (grepping for INCLUDE_ASM)
2. Name of the branch (for uniqueness-ish)
3. Name of the commit and PR

Which should be used for each of these 3?

Reply:
Idk how c++ projects name their .s files, but diff_label is the literal asm label of the function and display_name is what displays above the scratch (so alex often mentions c++ funcs have demangled names here). 

That being said, my option on these is that we should always be grepping for diff_label (1) (though this is the mangled name, so we should confirm files are named after this) and using display_name for any human-read things (2 and 3).